### PR TITLE
feat: add descriptive event title to dynatrace integration

### LINKED ISSUE #2414: Add descriptive event title to Dynatrace Integration
It was noted by Dynatrace that all of our events were just titled Deployment, it would be much better to have the feature name in the title at least.

image

https://www.dynatrace.com/support/help/platform-modules/cloud-automation/release-monitoring/version-detection-strategies#events-ingestion

### PR DESCRIPTION
**Pull request by matthewelwell**
## Changes

Adds the deployment title element to Dynatrace events. See #2414 for more details. 

## How did you test this code?

Updated existing unit tests.

### CODE REVIEW

File: api/integrations/common/wrapper.py [hunk 1]
```diff
@@ -18,8 +18,9 @@ def _track_event(self, event: dict) -> None:
     def track_event_async(self, event: dict) -> None:
         self._track_event(event)
 
-    @abstractstaticmethod
-    def generate_event_data(*args, **kwargs) -> None:
+    @staticmethod
+    @abstractmethod
```
Comment (matthewelwell) [inline api/integrations/common/wrapper.py:22]: This was just resolving a deprecation warning that is unrelated to this PR.

**Review by gagantrivedi — APPROVED** [review]
